### DISCUSSION TITLE
CopperSpice removed all internal documentation in their code

### DISCUSSION THREAD

**Discussion by ghost** [RolandHughes/copperspice#16] [General]
https://github.com/copperspice/copperspice/commit/44a9e9c045bf82c94ab8fd7ac0f7a81221d3378a

https://github.com/copperspice/copperspice/commit/e5a959d9f55ac2e497664ad31a54cc03aa2f5271

https://github.com/copperspice/copperspice/commit/548ac8fae9a7cfdc137c0ff451820028d06ca626

https://github.com/copperspice/copperspice/commit/b38b85a27c74e43131320dc537ead24fe48453aa

https://github.com/copperspice/copperspice/commit/2247c0e1f16539b656f71d17d3819853795937ae

https://github.com/copperspice/copperspice/commit/dd1c90cdc381449e2628c8afc280938507b920f1

**Comment by RolandHughes** [comment]
That should really help developers attempting to contribute . . . **NOT!**

I really don't care what they do. We are on a different road. This is about creating the application framework I've needed for every embedded system/medical device project I've ever worked on. It is also about dramatically easing the FDA 510K process.

```
/*!
 * \brief MessageSubscriber::MessageSubscriber
 * \param messageIds: The set of message IDs to subscribe to
 * \param parent: the parent object
 * Constructs the MessageSubsriber
 */

/*!
 * \brief VirtualKeyboard::withinLimits
 * \param tstString     String to be tested against limits
 * \returns bool        true if string is within limits and false if not
 *
 * Test string against maximum allowed length.
 */
```
Comments like that need to exist for every class, method, stand alone function, and constant. That stage will happen _after_ we get Servo implemented and before we go further down the roadmap. 

I'm not interested in maintaining backward compatibility with CopperSpice or continuing to be "just like CopperSpice" or viewed as "tweaked CopperSpice." This package will have massive architectural changes for the world in which I live.
